packages/app/src/components-styled/area-chart/area-chart.tsx

```diff
@@ -0,0 +1,382 @@
+          tooltipTop: nearestPoint.y,
+        });
+        setMarkerProps({
+          data: [nearestPoint],
+        });
+      }
+    },
+    [showTooltip, hideTooltip]
+  );
+
+  const bisect = useBisect(padding);
+
+  const onHover = useChartHover(
+    toggleHoverElements,
+    trendConfigs,
+    areaConfigs,
+    bisect
+  );
+
+  const bounds: ChartBounds = {
+    width: width - padding.left - padding.right,
+    height: height - padding.top - padding.bottom,
+  };
+
+  function getDate(x: TrendValue) {
+    return x.__date;
+  }
+
+  const timespanMarkerData = trendConfigs[0].values;
+
+  const dateSpanScale = useMemo(
+    () =>
+      scaleBand<Date>({
+        range: [0, xMax],
+        domain: timespanMarkerData.map(getDate),
+      }),
+    [xMax, timespanMarkerData]
+  );
+
+  const xScale = scaleTime({
+    domain: xDomain,
+    range: [0, bounds.width],
+  });
+
+  const yScale = scaleLinear({
+    domain: yDomain,
+    range: [bounds.height, 0],
+    nice: NUM_TICKS,
+  });
+
+  const scales = { xScale, yScale };
+
+  const handleHover = (
+    event: TouchEvent<SVGElement> | MouseEvent<SVGElement>
+  ) => onHover(event, scales);
+
+  return (
+    <Box position="relative">
+      {isDefined(valueAnnotation) && (
+        <ValueAnnotation>{valueAnnotation}</ValueAnnotation>
+      )}
+
+      <AreaChartGraph
+        trends={trendConfigs}
+        areas={areaConfigs}
+        bounds={bounds}
+        width={width}
+        height={height}
+        padding={padding}
+        scales={scales}
+        formatXAxis={formatXAxis}
+        formatYAxis={isPercentage ? formatYAxisPercentageFn : formatYAxisFn}
+        numTicks={6}
+        onHover={handleHover}
+      >
+        {divider &&
+          renderDivider(areaConfigs as any, divider, height, padding, xScale)}
+      </AreaChartGraph>
+
+      <Box
+        height={yMax}
+        width={xMax}
+        position="absolute"
+        top={padding.top}
+        left={padding.left}
+        style={{
+          pointerEvents: 'none',
+        }}
+      >
+        {markerProps && (
+          <Marker
+            {...markerProps}
+            showLine={false}
+            dateSpanWidth={dateSpanScale.bandwidth()}
+            height={height}
+            padding={padding}
+          />
+        )}
+      </Box>
+
+      {isDefined(tooltipData) && (
+        <Tooltip
+          bounds={{ right: width, left: 0, top: 0, bottom: height }}
+          x={tooltipLeft + padding.left}
+          y={tooltipTop + padding.top}
+        >
+          {formatTooltip(tooltipData, isPercentage)}
+        </Tooltip>
+      )}
+    </Box>
+  );
+}
+
+function renderDivider(
+  areas: AreaConfig<TimestampedTrendValue>[],
+  divider: DividerConfig,
+  height: number,
+  padding: ChartPadding,
+  xScale: ScaleTime<number, number>
+) {
+  const dates = areas.map((area) => area.values[0].__date);
+  dates.shift();
+
+  return dates.map((date) => {
+    const x = xScale(date);
+    return x !== undefined ? (
```

Comment:
nit; This is a bit easier to follow I think.

```suggestion
    return x === undefined ? null :
 
```